### PR TITLE
Change protobuf to betterproto

### PR DESCRIPTION
**Pull request by warittornc**

### CODE REVIEW

File: pyband/messages/cosmos/bank/v1beta1/msgs.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+from ....common import MessageWrapper
+from .....proto.cosmos.bank.v1beta1 import MsgMultiSend as MsgMultiSendProto
+from .....proto.cosmos.bank.v1beta1 import MsgSend as MsgSendProto
+
+try:
+    from .....proto.cosmos.base import v1beta1 as __base_v1_beta1__
+    from .....proto.cosmos.base.query import v1beta1 as __base_query_v1_beta1__
+except ImportError as ie:
+    raise ie
```
Comment (taobun) [inline pyband/messages/cosmos/bank/v1beta1/msgs.py:11]: Why need to have try/except if you still raise exception

File: setup.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
     author_email="dev@bandprotocol.com",
     url="https://github.com/bandprotocol/bandchain",
     keywords=["BAND", "BLOCKCHAIN", "ORACLE"],
-    install_requires=["bech32", "bip32", "ecdsa", "mnemonic", "grpcio", "protobuf"],
+    install_requires=["bech32==1.2.0", "bip32==0.0.8", "ecdsa==0.15", "mnemonic==0.19", "betterproto==2.0.0b5"],
```
Comment (taobun) [inline setup.py:19]: Do we need grpclib?

**Review by taobun — CHANGES_REQUESTED** [review]

File: pyband/messages/cosmos/bank/v1beta1/msgs.py (hunk 1 above)
Comment (warittornc) [reply pyband/messages/cosmos/bank/v1beta1/msgs.py]: changed to assert

File: setup.py (hunk 2 above)
Comment (warittornc) [reply setup.py:19]: its already in betterproto's dependency

File: pyband/transaction.py [hunk 3]
```diff
@@ -1,58 +1,56 @@
...
+
         account = client.get_account(sender)
-        if account:
+        if account is not None:
             self.account_num = account.account_number
             self.sequence = account.sequence
             return self
-        raise NotFoundError("Account doesn't exist")
+        else:
+            raise NotFoundError("Account doesn't exist")
```
Comment (taobun) [inline pyband/transaction.py:53]:
```
if account is None:
   raise(...)
self...
```

File: pyband/transaction.py [hunk 4]
```diff
@@ -130,17 +123,18 @@ def get_sign_message(self) -> bytes:
             "account_number": str(self.account_num),
             "chain_id": self.chain_id,
             "fee": {
-                a.replace("gasLimit", "gas"): b for a, b in json.loads(json_format.MessageToJson(self.fee)).items()
+                "amount": [fee_amount.to_dict() for fee_amount in self.fee.amount],
```
Comment (taobun) [inline pyband/transaction.py:126]: I recommend to rename this function to `get_sign_message_for_legacy_codec`

File: setup.py [hunk 5]
```diff
@@ -16,7 +16,15 @@
     author_email="dev@bandprotocol.com",
     url="https://github.com/bandprotocol/bandchain",
     keywords=["BAND", "BLOCKCHAIN", "ORACLE"],
-    install_requires=["bech32", "bip32", "ecdsa", "mnemonic", "grpcio", "protobuf"],
```
Comment (taobun) [inline setup.py:19]: I think we can remove `setup.py` anb `requirements.txt` (I prefer using poetry too)

**Review by taobun — CHANGES_REQUESTED** [review]
Looking good kub just a little comments

File: pyband/client.py [hunk 6]
```diff
@@ -1,80 +1,113 @@
...
+    def close(self) -> None:
+        self.__channel.close()
+
+    @classmethod
+    def from_endpoint(cls, grpc_endpoint: str, port: int, insecure: bool = False):
+        return cls(
+            Channel(
+                host=grpc_endpoint,
+                port=port,
+                ssl=False if not insecure else True,
```
Comment (taobun) [inline pyband/client.py:58]: 👀

**Review by taobun — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+python-dateutil = "2.8.2"
+bech32 = "1.2.0"
+bip32 = "0.0.8"
+ecdsa = "0.15"
+mnemonic = "0.19"
+betterproto = "2.0.0b5"
+grpclib = "0.4.1"
+ledgerblue = "0.1.42"
+
+
```
Comment (taobun) [inline pyproject.toml:33]: https://python-poetry.org/docs/managing-dependencies/#dependency-groups add testing dependency here and update test ci to install via poetry

File: examples/ledger_example.py [hunk 8]
```diff
@@ -1,17 +1,16 @@
...
-from pyband.proto.oracle.v1.tx_pb2 import MsgEditDataSource
-from pyband.proto.cosmos.tx.signing.v1beta1 import signing_pb2 as tx_sign
-from google.protobuf.json_format import MessageToJson
+from pyband.messages.oracle.v1 import MsgEditDataSource
+from pyband.proto.cosmos.base.v1beta1 import Coin
+from pyband.proto.cosmos.tx.signing.v1beta1 import SignMode
+from pyband.wallet import Ledger
 
 
 def main():
```
Comment (taobun) [inline examples/ledger_example.py:10]: This should be async def ?

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**